File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>

/**
 * Calculate min number of coins that are required to give
 * customer change.
 * 
 * @return int
 */
int main(void)
{
    // get change due from user
    float changef = 0;
    
    do {
        printf("How much change is due? e.g. $ 0.84\n");
        changef = GetFloat();
    } while ( !(changef > 0) );
    
    // Initiate change_due and coins variables
    int change_due = changef * 1000;
    int coins = 0;

    // Return max quaters (25¢)
    coins = coins + (change_due / 250);
    change_due = (change_due % 250);

    // Return max dimes (10¢)
    coins = coins + (change_due / 100);
    change_due = (change_due % 100);

    // Return max nickels (5¢)
    coins = coins + (change_due / 50);
    change_due = (change_due % 50);

    // Return max pennies (1¢)
    coins = coins + (change_due / 10);
    change_due = (change_due % 10);

    // return coin count to user    
    printf("%d\n", coins);
}

File: pset6/greedy.py
def main():
    # init change due var
    changef = 0;
    
    # get change due from user
    while True:
        changef = float(input("How much change is due? e.g. $0.84 : $"))
        if changef > 0:
            break;
    
    # init change_due and coins vars
    change_due = int(changef * 100)
    coins = 0
    coin_list = [25, 10, 5, 1]
    
    # loop through each coin
    for i in coin_list:
        # calculate max coins that can be deducted for this coin
        coins = coins + (change_due // i)
        # update change due
        change_due = (change_due % i)
    
    # return coin count to user
    print(coins)
    
# call main
if __name__ == "__main__":
    main()
    
    